documentation/cookbook.qmd

```diff
@@ -0,0 +1,266 @@
+
+## Basics
+
+### Recipe: Defining a Simple Pass/Fail Task
+The most fundamental task type is one that asserts a condition and returns nothing. If all assertions pass, the task succeeds; otherwise, it fails. This is perfect for simple Q&A checks.
+
+```python
+import kaggle_benchmarks as kbench
+
+@kbench.task(name="check_capital")
+def check_capital(llm):
+    # 1. Prompt the model
+    response = llm.prompt("What is the capital of France?")
+
+    # 2. Assert the correctness of the response
+    # We use a regex for robust, case-insensitive matching.
+    kbench.assertions.assert_contains_regex(
+        r"(?i)Paris", response, expectation="Model should answer Paris."
+    )
+
+check_capital.run(kbench.llm)
+```
+[See Example: Assertions Notebook](https://www.kaggle.com/code/limakaggle/kaggle-benchmark-cookbook-assertions)
+
+### Recipe: Returning a Numerical Score
+To track granular metrics like accuracy or a specific score, your task should return a value. You **must** add a return type annotation (e.g., `-> float`) so the leaderboard knows how to interpret the result.
+
+```python
+@kbench.task(name="math_score")
+def math_score(llm) -> float:
+    # ... perform complex logic or multiple checks ...
+    score = 0.85
+    return score
+```
+**Supported Return Types:**
+- `bool`: Pass/Fail (True/False)
+- `float` / `int`: A numerical score (e.g., 0.85)
+- `tuple[int, int]`: A count (e.g., `(8, 10)` for 8 out of 10 passed)
+- `tuple[float, float]`: A value with confidence interval (e.g., `(0.85, 0.05)`)
+
+[See Example: Return Types Notebook](https://www.kaggle.com/code/limakaggle/kaggle-benchmark-cookbook-return-types)
+
+### Recipe: Enforcing Structured Output with Schemas
+You can force the LLM to return data in a specific structure by providing a `schema` to the `prompt` method. This is essential for tasks that require structured output.
+
+```python
+from dataclasses import dataclass
+
+@dataclass
+class MovieReview:
+    sentiment: str
+    score: int
+
+@kbench.task(name="analyze_review")
+def analyze_review(llm):
+    # The model will return an instance of MovieReview, not a string.
+    result = llm.prompt(
+        "Analyze this review: 'Fantastic movie!'",
+        schema=MovieReview
+    )
+    print(f"Score: {result.score}, Sentiment: {result.sentiment}")
+```
+[See Example: Structured Output Notebook](https://www.kaggle.com/code/limakaggle/kaggle-benchmark-cookbook-structured-output)
+
+### Recipe: Publishing Your Task to the Leaderboard
+Currently our leaderboard only supports a single task per notebook (This may change in the future). To submit your benchmark to a Kaggle leaderboard, you need to designate one "main" task output using the `%choose` magic command.
+
+1.  **Define your tasks** in the notebook.
+2.  **Select the main task** in the very last cell using `%choose`.
```

Comment:
Do we need to add support for this to the AST parser or did we already do that?

Reply:
Currently it's implemented as a magic in IPython, which is pretty flexible. I don' think we have any AST level parsing yet since it's not clear to me the best way for user to choose a main task on AST level? Or shall we add a better UI here?